### PR TITLE
Make cluster details > worker nodes tab aware of permissions for a read-only user

### PR DESCRIPTION
**Pull request by kuosandys**
Towards https://github.com/giantswarm/roadmap/issues/551.

This PR makes the cluster details > worker nodes tab aware of permissions, and adapts the UI for a read-only user in the following ways:

### Disable "Add node pool" button and adapt messages:

Empty state:
Screen Shot 2021-12-07 at 17 01 38

With node pools:
Screen Shot 2021-12-06 at 17 51 41

Hovering over the disabled buttons also displays the cursor in the `not-allowed` style.

### Disable node pool description editing and adapt tooltip message:
Screen Shot 2021-12-07 at 14 07 17

Hovering over the description also displays the cursor in the `not-allowed` style.

### Disable node pool actions menu items and display tooltips on hover:
Screen Shot 2021-12-06 at 17 20 29 Screen Shot 2021-12-06 at 17 20 37

Hovering over the disabled menu items also displays the cursor in the `not-allowed` style.

### Display warning messages in CLI guides:

For modifying node pools:
Screen Shot 2021-12-07 at 14 02 38

For deleting node pools:
Screen Shot 2021-12-07 at 14 02 48

---
### How to test the changes of this PR
An easy way to test is by impersonating a user with a rolebinding to the `read-all` cluster role within an organization namespace. This can be done from `happa` or with `kubectl`.
#### From `happa`:
1. Navigate to the Organization details page by clicking on "Organizations" in the nav bar and clicking on your organization in the following page. Select the "Access Control" tab.
3. Select the `read-all` role from the role list in the left panel; under the "Users" heading in the right panel, click on the "Add" button.
4. Enter the name for a test user (e.g. "test-user") and click "OK".
Screen Shot 2021-12-07 at 16 45 31

#### With `kubectl`:
`kubectl create rolebinding test-binding --user=test-user --clusterrole=read-all --namespace=org-giantswarm`

Substitute the user and organization namespace fields as you wish.

Once the rolebinding has been created, start local development for `happa` by following the instructions in the `README` and check-out this branch. Then, login to `happa` as an admin user and navigate to the Experiments page. This can be found in the dropdown menu at the top-right corner of the page.
In the Impersonation section, input the user specified in the rolebinding in the "Username" field. Click "Set" to impersonate the user.
Screen Shot 2021-12-07 at 15 22 45

### CODE REVIEW

File: src/components/MAPI/clusters/ClusterList/ClusterListItem.tsx [hunk 1]
```diff
@@ -144,41 +144,15 @@ const ClusterListItem: React.FC<IClusterListItemProps> = ({
...
-      default:
-        return null;
-    }
-  }, [
-    cluster,
-    hasReadPermissionsForNpInDefaultNamespace,
-    hasReadPermissionsForNpInOrgNamespace,
-    orgNamespace,
-  ]);
+  const { canList: canListNodePools, canGet: canGetNodePools } =
```
Comment (kuosandys) [inline src/components/MAPI/clusters/ClusterList/ClusterListItem.tsx:147]: This refactors how we determine permissions for node pool resources: we check the user's permissions for node pools in the same namespace as the cluster's namespace.

e.g. If the cluster is in the `org-giantswarm` namespace, we check if the user has permissions for node pool resources in the `org-giantswarm` namespace.

File: src/components/MAPI/workernodes/permissions/usePermissionsForNodePools.ts [hunk 2]
```diff
@@ -132,6 +132,13 @@ export function usePermissionsForNodePools(
           'create',
           'infrastructure.cluster.x-k8s.io',
           'azuremachinepools'
+        ) &&
```
Comment (kuosandys) [inline src/components/MAPI/workernodes/permissions/usePermissionsForNodePools.ts:135]: Missed this before - `Sparks` is also a node pool resource

File: src/components/UI/Controls/Button/index.tsx [hunk 3]
```diff
@@ -175,7 +175,7 @@ const Button = React.forwardRef<
               buttonStyle !== ButtonStyle.Link
             }
             label={children}
-            disabled={disabled || loading}
+            disabled={disabled || loading || unauthorized}
```
Comment (kuosandys) [inline src/components/UI/Controls/Button/index.tsx:178]: Update `Button` logic - if a button displays that a user lacks permissions (unauthorized), it should also be shown as disabled.

**Review by marians — APPROVED** [review]
Works as expected 👍

**Pull request merged**